src/hooks/useEventListener/useEventListener.ts

```diff
@@ -1,10 +1,10 @@
 /* eslint-disable  @typescript-eslint/unified-signatures, @typescript-eslint/no-explicit-any */
-import { useEffect, type RefObject } from 'react';
-import { unref } from '../../utils/unref/unref.js';
+import { useEffect } from 'react';
+import { unref, type Unreffable } from '../../utils/unref/unref.js';
 import { useRefValue } from '../useRefValue/useRefValue.js';
 
-export function useEventListener<T extends EventTarget>(
-  targetOption: RefObject<T> | T | null | undefined,
+export function useEventListener<T extends EventTarget | null>(
+  targetOption: Unreffable<T>,
```

Comment:
I think using `Unreffable<T>` and adding `<T extends EventTarget | null>` is the correct way to support refs that can be null.